httpql/src/test/java/com/hubspot/httpql/model/EntityComplexJoinDescriptor.java

```diff
@@ -0,0 +1,30 @@
+package com.hubspot.httpql.model;
+
+import org.jooq.Field;
+import org.jooq.impl.DSL;
+
+import com.hubspot.httpql.QuerySpec;
+import com.hubspot.httpql.ann.desc.JoinDescriptor;
+import com.hubspot.httpql.impl.JoinCondition;
+import com.hubspot.httpql.internal.FilterEntry;
+
+public class EntityComplexJoinDescriptor implements JoinDescriptor {
+  private static final String JOIN_TABLE_NAME = "join_tbl";
+
+  @Override
+  public Field<?> getField(FilterEntry filterEntry) {
+    return DSL.nvl(DSL.field(DSL.name(JOIN_TABLE_NAME, filterEntry.getQueryName())), 0);
```

Comment:
you could static import some of these DSL methods to make this class a bit easier to read.